src/chainstate/burn/operations/leader_block_commit.rs

```diff
@@ -81,15 +83,19 @@ fn u16_from_be(bytes: &[u8]) -> Option<u16> {
     }
 }
 
-impl LeaderBlockCommitOp<BitcoinPublicKey> {
-    fn parse_data(data: &Vec<u8>) -> Option<(BlockHeaderHash, VRFSeed, u32, u16, u32, u16, Vec<u8>)> {
+impl<AddrType, PubkeyType> LeaderBlockCommitOp<AddrType, PubkeyType>
+where
+    AddrType: Address,
+    PubkeyType: PublicKey
+{
+    fn parse_data(data: &Vec<u8>) -> Option<(BlockHeaderHash, VRFSeed, u16, u16, u16, u16, u32, Vec<u8>)> {
```

Comment:
It'd be better to return a typed object here rather than a really long tuple -- these kinds of long tuples can lead to errors as code changes.

Reply:
Sure, I can do that.